src/views/List.jsx

```diff
@@ -1,18 +1,46 @@
+import { useState } from 'react';
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	const [filter, setFilter] = useState('');
+
+	const clickHandler = (event) => {
+		event.preventDefault();
+		setFilter('');
+	};
+
 	return (
 		<>
-			<p>
-				Hello from the <code>/list</code> page!
-			</p>
+			<form>
+				<label htmlFor="itemInput">Filter Items: </label>
+				<input
+					type="text"
+					id="itemInput"
+					name="itemInput"
+					placeholder="Start typing here..."
+					value={filter}
+					onChange={(event) => setFilter(event.target.value)}
+				/>
+				{filter && (
+					<span>
+						<button type="reset" onClick={clickHandler}>
+							X
+						</button>
+					</span>
+				)}
+			</form>
 			<ul>
 				{data.length > 0 ? (
-					<ul>
-						{data.map((item) => {
-							return <ListItem name={item.name} key={item.id} />;
-						})}
-					</ul>
+					<li>
+						{data
+							.filter(
+								(item) =>
+									item.name.includes(filter.toLowerCase()) || filter === '',
```

Comment:
My preference here is that instead of saving the item with a lowercase name in `src/api/firebase.js`, instead it might be better to save it as it was entered and then lowercase it here in the filter:

```suggestion
									item.name.toLowerCase().includes(filter.toLowerCase()) || filter === '',
```

Either way works!

Reply:
Gotcha. I wasn't sure if it was best practice to just store the data as is or change it before it is stored. I've now implemented your suggestion. Thank you!